CMakeLists.txt

```diff
@@ -3,47 +3,63 @@ cmake_minimum_required(VERSION 3.4)
 
 project(PopSift VERSION 1.0.0)
 
-OPTION(PopSift_BUILD_EXAMPLES "Build PopSift applications." ON)
+OPTION(PopSift_BUILD_EXAMPLES "Build PopSift applications."  ON)
+OPTION(USE_NVTX_PROFILING     "Use CUDA NVTX for profiling." OFF)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${PROJECT_SOURCE_DIR}/cmake")
 
+# set(CMAKE_BUILD_TYPE Debug)
 if(NOT CMAKE_BUILD_TYPE)
   set(CMAKE_BUILD_TYPE Release)
   message(STATUS "Build type not set, building in Release configuration")
 else()
   message(STATUS "Building in ${CMAKE_BUILD_TYPE} configuration")
 endif()
 
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_DEBUG} -O3")
-set(CMAKE_C_FLAGS_RELEASE   "${CMAKE_C_FLAGS_DEBUG}   -O3")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
+set(CMAKE_C_FLAGS_RELEASE   "${CMAKE_C_FLAGS_RELEASE}   -O3")
 
-#set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DGRIFF_DEBUG")
-#set(CMAKE_C_FLAGS_DEBUG   "${CMAKE_C_FLAGS_DEBUG}   -DGRIFF_DEBUG")
+# set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -G")
+# set(CMAKE_C_FLAGS_DEBUG   "${CMAKE_C_FLAGS_DEBUG}   -G")
+
+find_package(Boost 1.53.0 REQUIRED COMPONENTS system thread)
+if(WIN32)
+ADD_DEFINITIONS("-DBOOST_ALL_NO_LIB")
+LINK_DIRECTORIES(Boost_LIBRARRY_DIR_DEBUG)
+LINK_DIRECTORIES(Boost_LIBRARRY_DIR_RELEASE)
+
+endif()
 
 find_package(CUDA 7.0)
 
 if(NOT CUDA_FOUND)
   message(FATAL_ERROR "Could not find CUDA >= 7.0")
 endif()
 
+if(USE_NVTX_PROFILING)
+  message(STATUS "PROFILING CPU CODE: NVTX is in use")
+endif(USE_NVTX_PROFILING)
+
 set(CUDA_SEPARABLE_COMPILATION ON)
 
-# The following if should not be necessary, but apparently there is a bug in FindCUDA.cmake that
-# generate an empty string in the nvcc command line causing the compilation to fail.
-# see https://gitlab.kitware.com/cmake/cmake/issues/16411
-if(CMAKE_BUILD_TYPE STREQUAL "Debug")
-  message(STATUS "Building in debug mode")
-  set(CUDA_NVCC_FLAGS_DEBUG   "${CUDA_NVCC_FLAGS_DEBUG};-G;-g")
-else()
-  message(STATUS "Building in release mode")
-  set(CUDA_NVCC_FLAGS_RELEASE "${CUDA_NVCC_FLAGS_RELEASE};-O3;-DNDEBUG")
-  # set(CUDA_NVCC_FLAGS_RELEASE "${CUDA_NVCC_FLAGS_RELEASE};-G;-g;-DNDEBUG")
-endif()
+# set(CUDA_NVCC_FLAGS_DEBUG   "${CUDA_NVCC_FLAGS_DEBUG};-G;-g")
+if(APPLE)
+elseif(NOT WIN32)
```

Comment:
? question about NOT WIN32 or about commenting -G -g?
The first is an artifact of merging 2 ifs.
The second is because -g is added automatically in Debug mode, so only -G needs to added explicitly.